src/dvc_objects/db.py

```diff
@@ -45,10 +43,22 @@ def __hash__(self):
         return hash((self.fs.protocol, self.path))
 
     def exists(self, oid: str) -> bool:
-        return self.fs.exists(self.oid_to_path(oid))
+        return self.fs.isfile(self.oid_to_path(oid))
 
     def exists_prefix(self, short_oid: str) -> str:
-        ret = [oid for oid in self.all() if oid.startswith(short_oid)]
+        path = self.oid_to_path(short_oid)
+        if len(short_oid) <= 2:
+            raise ValueError(short_oid, [])
+
+        if self.exists(path):
+            return short_oid
```

Comment:
This seems to be changing the existing behaviour. That directory might be empty.